ui/components/imageWorkspace/src/components/Features/TextInputFeature.svelte

```diff
@@ -43,12 +55,21 @@
   {#if isEditing}
     <Input
       value={textFeature.value}
-      on:change={(e) => onTextInputChange(e.currentTarget.value, textFeature.name)}
-      on:input={() => (isSaved = false)}
       type={inputType === "str" ? "text" : "number"}
       step={inputType === "int" ? "1" : "any"}
+      list="scene_availableValues_{textFeature.name}"
+      on:change={(e) => onTextInputChange(e.currentTarget.value, textFeature.name)}
+      on:input={() => (isSaved = false)}
       on:keyup={(e) => e.stopPropagation()}
     />
+    <datalist id="scene_availableValues_{textFeature.name}">
+      {#if "scene" in $itemFeaturesAvailableValues && textFeature.name in $itemFeaturesAvailableValues.scene}
```

Comment:
I think we could narrow down `FeaturesAvailableValues` to something more explicitif 

```
interface FeaturesAvailableValues {
  scene: Record<string, Array<string>>;
  object: Record<string, Array<string>>;
}
```

that would make your component type safer and would avoid these type check everywhere. That would also make the code more readable. 
If the future we add more categories we will either refacto to something more generic or add the new category to the type.